### ISSUE TITLE
Question: Is it possible to use a Pi to create a Sonos Speaker?

### ISSUE THREAD

**Issue by warrenbuckley** [SoCo/socos#58]
Hello,
Sorry to post & use your GitHub issue tracker but I have yet to be accepted into the Google Group to discuss & ask this question.

With the SoCo library that you guys have developed, would it be possible to create a Sonos Speaker to go on the network from say a Raspberry Pi.

I currently have a Play:1 & Play:3 in our house, but the wife likes the look of vintage radios to go into the kitchen. So would it be possible to use your library with a Pi & connect it/wire it up to an old radio speaker in order to create a nice looking Sonos speaker for her?

From what I can tell your library allows to manage Sonos speakers, queues, volume control etc which you have managed to do by looking at network traffic from what I understand, so I am curious to see if this same knowledge could be applied in order of creating a speaker/endpoint that understands the commands being sent to it to act like a Sonos speaker.

Again sorry for the question here on your issue tracker

**Comment by KennethNielsen** [comment]
Quick answer; no.

Slightly longer answer: SoCo by nature is a Sonos client and what you want is a server. Of course some knowledge about the communication is contained in SoCo, but creating a server would be a entirely different project. Also (and I'm just guessing here) I can imagine several problems with such a project. First of, at least if you want your speaker recognized by the other speakers/official client, then you will need intimate knowledge of the discovery protocol, which I'm not sure we have. Second, while the authentication with music services from the client side is specified/standard technologies, I'm not sure that is enough information to implement the server part of that.

Basically I think your best bet to satisfy the requirement you mention is, to use a connect and wire the output from that into an input on the radio, or maybe, if you are building something that looks like a vintage radio, you can hide the connect inside? It is however somewhat more expensive that a Raspberry Pi.

**Comment by warrenbuckley** [comment]
Thanks for the answer @KennethNielsen I just assumed that with you understanding how a client works that you may know how a server works. But it seems like I need to do some work to hide the `connect` into the radio for my wife then.

Thanks for the advice and keep up the good work with the project

**Comment by KennethNielsen** [comment]
@warrenbuckley Well, I'm not saying that it may not be possible and that SoCo probably will not be as much help.

**Comment by warrenbuckley** [comment]
No problem at all. I was just hoping that you guys had intimate knowledge of both client & server already so I could get started playing, but realistically I will probably get a connect & wire it into vintage radio as needed
